leaves/commits.go

```diff
@@ -0,0 +1,229 @@
+package leaves
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/gogo/protobuf/proto"
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/hercules.v9/internal/core"
+	"gopkg.in/src-d/hercules.v9/internal/pb"
+	items "gopkg.in/src-d/hercules.v9/internal/plumbing"
+	"gopkg.in/src-d/hercules.v9/internal/plumbing/identity"
+	"gopkg.in/src-d/hercules.v9/internal/yaml"
+)
+
+// CommitsAnalysis extracts statistics for each commit
+type CommitsAnalysis struct {
+	core.NoopMerger
+	core.OneShotMergeProcessor
```

Comment:
`core.OneShotMergeProcessor` can be safely excluded. We check if the commit is a merge commit in `Consume`, so this feature (to process each merge only once) is not required.